server/app/views/admin/programs/ProgramIndexViewV2.java

```diff
@@ -232,108 +342,78 @@ private static Instant extractLastUpdated(
+    String editLinkId = "program-edit-link-" + program.id();
+    if (isActive) {
+      editLink = controllers.admin.routes.AdminProgramController.newVersionFrom(program.id()).url();
+      editLinkId = "program-new-version-link-" + program.id();
+    }
 
-    if (draftProgram.isPresent()) {
-      String editLink =
-          controllers.admin.routes.AdminProgramController.edit(draftProgram.get().id()).url();
-
-      return new LinkElement()
-          .setId("program-edit-link-" + draftProgram.get().id())
-          .setHref(editLink)
-          .setText(editLinkText)
-          .setStyles(Styles.MR_2)
-          .asAnchorText();
-    } else if (activeProgram.isPresent()) {
-      String newVersionLink =
-          controllers.admin.routes.AdminProgramController.newVersionFrom(activeProgram.get().id())
-              .url();
+    ContainerTag button =
+        makeSvgTextButton("Edit", Icons.EDIT_SVG_PATH)
+            .withId(editLinkId)
+            .withClasses(AdminStyles.TERTIARY_BUTTON_STYLES);
+    return isActive
+        ? toLinkButtonForPost(button, editLink, request)
+        : asRedirectButton(button, editLink);
+  }
 
-      return new LinkElement()
-          .setId("program-new-version-link-" + activeProgram.get().id())
-          .setHref(newVersionLink)
-          .setText(newVersionText)
-          .setStyles(Styles.MR_2)
-          .asHiddenForm(request);
-    } else {
-      // obsolete or deleted, no edit link, empty div.
-      return div();
-    }
+  private Tag renderManageTranslationsLink(ProgramDefinition program) {
+    String linkDestination =
+        routes.AdminProgramTranslationsController.edit(
+                program.id(), LocalizedStrings.DEFAULT_LOCALE.toLanguageTag())
+            .url();
+    ContainerTag button =
+        makeSvgTextButton("Manage translations", Icons.LANGUAGE_SVG_PATH)
+            .withId("program-translations-link-" + program.id())
+            .withClass(AdminStyles.TERTIARY_BUTTON_STYLES);
+    return asRedirectButton(button, linkDestination);
   }
 
-  private Tag maybeRenderManageTranslationsLink(Optional<ProgramDefinition> draftProgram) {
-    if (draftProgram.isPresent()) {
-      String linkText = "Manage Translations →";
-      String linkDestination =
-          routes.AdminProgramTranslationsController.edit(
-                  draftProgram.get().id(), LocalizedStrings.DEFAULT_LOCALE.toLanguageTag())
-              .url();
-      return new LinkElement()
-          .setId("program-translations-link-" + draftProgram.get().id())
-          .setHref(linkDestination)
-          .setText(linkText)
-          .setStyles(Styles.MR_2)
-          .asAnchorText();
-    } else {
-      return div();
+  private Optional<Tag> maybeRenderViewApplicationsLink(
+      ProgramDefinition activeProgram, CiviFormProfile userProfile) {
+    // TODO(#2582): Determine if this has N+1 query behavior and fix if
+    // necessary.
+    boolean userIsAuthorized = true;
```

Comment:
It makes me feel better if authorization-type bools default to false, and get set to true only if the user is authorized.

Reply:
Works for me. Rather than defaulting to false, I explicitly set it in each case. Also note that if I tried defaulting to false, that would cause an empty catch block, which would trigger a warning :).